src/cpp/runner/main.cpp

```diff
@@ -9,26 +10,144 @@
 
 using namespace std;
 
-/*GMMOutput readInputData() {
+GMMInput read_input_data(const string& input_file, const bool replicate_point)
+{
+    GMMInput input;
+
+    // Read instance
+    read_gmm_instance(input_file, &input.d, &input.k, &input.n,
+        input.alphas, input.means, input.icf, input.x, input.wishart, replicate_point);
+
+    return input;
+}
+
+typedef  void (ITest<GMMInput, GMMOutput>::* test_member_function) (int);
+
+inline void call_member_function(unique_ptr<ITest<GMMInput, GMMOutput>>& ptr_to_object, const test_member_function ptr_to_member, const int arg1) {
```

Comment:
There is said [here ](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es31-dont-use-macros-for-constants-or-functions )that we should avoid macros as much as possible. 
So, I propose a template.

Reply:
Can a template help with enforcing inlining? I agree, that we should avoid macros as much as possible. I argue, that here is the place where we actually need one. Unless, we are willing to go with the direct C++ syntax.

Reply:
Given how "repeats" works, any overhead for calling this will not affect timing, so it's fine to simply remove the "inline".   (It will with very high probability be inlined anyway).

Reply:
Yes, you're right. Since this will be called only once per time measurement (and not `repeats` times) it won't produce any significant effect on the benchmark result. Probably, we could even have gotten away with std::function/bind. No reason to do it now, that we've got pointers working, though.